### PR TITLE
fix assigning cast intent task to all group members when order closes

### LINKED ISSUE #285: task to 'add items to order' doesn't show for group members
it does appear for the order admin, but not for other members of the group

### PR DESCRIPTION
**Pull request by iainkirkpatrick**
closes #285

### CODE REVIEW

**Review by ahdinosaur — APPROVED** [review]
shot 🏀

**Comment by ahdinosaur** [comment]
oops, above review was meant for https://github.com/root-systems/cobuy/pull/287

File: tasks/services/works.js [hunk 1]
```diff
@@ -31,31 +31,34 @@ function completeOrderSetupTaskWork (hook) {
 
 function createCastOrderIntentTaskPlan (hook) {
   const taskPlans = hook.app.service('taskPlans')
+  const orders = hook.app.service('orders')
   const relationships = hook.app.service('relationships')
   const taskRecipeId = taskRecipes.castIntent.id
   const completeOrderTaskPlanId = hook.data.taskPlanId
+  let params
```
Comment (ahdinosaur) [inline tasks/services/works.js]: what's up with this?

File: tasks/services/works.js [hunk 2]
```diff
@@ -31,31 +31,34 @@ function completeOrderSetupTaskWork (hook) {
...
-    const groupId = completeOrderTaskPlan.params.consumerAgentId
     const orderId = completeOrderTaskPlan.params.orderId
-    const supplierAgentId = completeOrderTaskPlan.params.supplierAgentId
-    const params = { orderId, consumerAgentId: groupId, supplierAgentId }
-
+    return orders.get(orderId)
+  })
+  .then((order) => {
+    const { id, consumerAgentId, supplierAgentId, adminAgentId } = order
+    params = { orderId: id, consumerAgentId, supplierAgentId, adminAgentId }
```
Comment (ahdinosaur) [inline tasks/services/works.js]: why not pass this down the promise pipeline? it seems real sketchy (and prone to race conditions) to pass it through a global variable.

you could instead do:

```js
.then((order) => {
  const params = {}
  return Promise.all([
    Promise.resolve(params),
    relationships.find({})
  ])
})
.then(([params, relationships]) => {
})
```

**Review by ahdinosaur — CHANGES_REQUESTED** [review]
i'm sketched out by the params variable. might be totally innocuous and legit, but concerned about future race conditions or spaghetti.

**Comment by iainkirkpatrick** [comment]
good call, will change :)

**Pull request merged**